dev-packages/localization-manager/src/deepl-api.ts

```diff
@@ -47,6 +52,22 @@ export async function deepl(
     return mergedResponse;
 }
 
+async function postWithRetry(url: string, buffer: Buffer, attempt: number): Promise<DeeplResponse> {
+    try {
+        await rateLimiter.removeTokens(attempt);
```

Comment:
Could this error (https://github.com/eclipse-theia/theia/actions/runs/13050402257/job/36409193760#step:7:151) be related to this line? Shouldn't we be removing only one token per attempt?

Reply:
Yes, but we should use a backoff algorithm, so we don't run into this every 100ms.

Reply:
Ah, I see. So the error is just because we tried to remove more tokens than there were, even if full, and not because it got negative. Now it works: https://github.com/eclipse-theia/theia/pull/14796